### ISSUE TITLE
Nodes get deselected onMouseUp

### ISSUE THREAD

**Issue by kputnins** [antvis/G6#2922]
- [x] I have searched the [issues](https://github.com/antvis/g6/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/onmouseup-deselect-vwxdq?file=/src/index.js)

### Steps to reproduce
See link to [codesandbox](https://codesandbox.io/s/onmouseup-deselect-vwxdq?file=/src/index.js)

1. Use built in behavior `'click-select'`
2. Select a node or multiple nodes
3. Start mouse click on some other `HTML` element
4. Release mouse click on the G6 `<canvas />` board

Expected - Nodes still selected
Actual - Nodes de-selected

| Environment | Info |
|---|---|
| g6 | 4.3.2 |
| System | Windows |
| Browser | Chrome, Firefox, Edge |

---
Basically a random click that started anywhere but got released on the board will de-select the selected nodes. In the application we're building this provides to be an obstacle as on top of the the G6 canvas we have other React components (topbar, sidebar, etc), but clicks originating there should not affect the state of the graph

In the [documentation](https://g6.antv.vision/en/docs/manual/middle/states/defaultBehavior/#click-select) I did not find precise description how the "de-selection" works or what is the outcome in this case.

`shouldBegin(e)` gets triggered only on *selecting* nodes, not *de-selecting*, so does not help me here

**Comment by Yanyan-Wang** [comment]
solved by v4.8.0
